Extension/src/LanguageServer/configurations.ts

```diff
@@ -243,6 +243,33 @@ export class CppProperties {
         vscode.workspace.onDidChangeTextDocument((e: vscode.TextDocumentChangeEvent) => {
             if (e.document.uri.fsPath === settingsPath) {
```

Comment:
This settingsPath should have the location of the c_cpp_properties.json, so you should be able to call vscode.workspace.getWorkspaceFolder to check if it's outside a workspace folder.

Reply:
Great tip, thanks.